docs/data/tools_en.md

```diff
@@ -0,0 +1,152 @@
+- `--crop_size`: For datasets that support image cropping, specify the size of the cropped image block. Example: `--crop_size 256`.
+- `--crop_stride`: For datasets that support image cropping, specify the step size of the sliding window during cropping. Example: `--crop_stride 256`.
+- `--seed`: Random seed. It can be used to fix the pseudo-random number sequence generated by the random number generator, so as to obtain a fixed dataset partitioning result. Example: `--seed 1919810`
+- `--ratios`: For datasets that support random subset partitioning, specify the sample ratios of each subset that needs to be partitioned. Example: `--ratios 0.7 0.2 0.1`.
+
+You can refer to [this document](https://github.com/PaddlePaddle/PaddleRS/blob/develop/docs/intro/data_prep.md) to see which preprocessing scripts for datasets are provided by PaddleRS.
+
+### extract_ms_patches
+
+The main function of `extract_ms_patches.py` is to extract image patches containing objects of interest at different scales from the entire remote sensing image using a quadtree. The extracted image patches can be used as training samples for models. The usage is as follows:
+
+```shell
+python extract_ms_patches.py --im_paths {one or more input image paths} --mask_path {Ground-truth label path} [--save_dir {output directory}] [--min_patch_size {minimum patch size}] [--bg_class {background class ID}] [--target_class {target class ID}] [--max_level {maximum level of scale}] [--include_bg] [--nonzero_ratio {threshold of the ratio of nonzero pixels}] [--visualize]
+```
+
+Among them:
+
+- `im_paths`: Path of the source image(s), can specify multiple paths.
+- `mask_path`: Path to the Ground-truth label.
+- `save_dir`: Path to the directory to save the split result. Default is `output`.
+- `min_patch_size`: The minimum size of the extracted image block (in terms of the number of pixels in the length/width of the image block covered by the leaf nodes in the quadtree). Default is `256`.
+- `bg_class`: The class ID of the background class. Default is `0`.
+- `target_class`: The class ID of the target class. If it is `None`, it means that all classes except the background class are target classes. Default is `None`.
+- `max_level`: The maximum level of scale to retrieve. If it is `None`, it means that there is no limit to the level. Default is `None`.
+- `include_bg`: If specified, also save the image blocks that only contain the background class and do not contain the target class.
+- `--nonzero_ratio`: Specify a threshold. For any source image, if the ratio of nonzero pixels in the image block is less than this threshold, the image block will be discarded. If it is `None`, no filtering will be performed. Default is `None`.
+- `--visualize`: If specified, after the program is executed, the image `./vis_quadtree.png` will be generated, which contains the visualization result of the node situation in the quadtree. An example is shown in the following figure:
```

Comment:
`contains the visualization result of the node situation` -> `visualizes the nodes`